alerter.py

```diff
@@ -1,25 +1,31 @@
-alert_failure_count = 0
-
+alert_failure_count =0
+Threshold_temp = 100
 def network_alert_stub(celcius):
     print(f'ALERT: Temperature is {celcius} celcius')
     # Return 200 for ok
+    if (celcius <=100 and celcius>=0):
+        return 200
     # Return 500 for not-ok
-    # stub always succeeds and returns 200
-    return 200
+    else:
+        return 500
 
 def alert_in_celcius(farenheit):
+    global alert_failure_count
+    alert_failure_count=0
     celcius = (farenheit - 32) * 5 / 9
     returnCode = network_alert_stub(celcius)

```

Comment:
stub/test code and production code is the same file. They should be kept separate so that it helps in future maintainability and also test code can be easily excluded during shipment of software to field.

The production code should not be aware of test code or test stub. It should be abstracted by the use of interface or function pointer.